### PR TITLE
Removed Saga dependency

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

File: rendering_resource_manager_service/config/models.py [hunk 1]
```diff
@@ -39,6 +39,12 @@ class RenderingResourceSettings(models.Model):
     modules = models.CharField(max_length=4096, default='')
     process_rest_parameters_format = models.CharField(max_length=1024, default='')
     scheduler_rest_parameters_format = models.CharField(max_length=1024, default='')
+    project = models.CharField(max_length=256)
+    queue = models.CharField(max_length=256)
+    exclusive = graceful_exit = models.BooleanField(default=True)
```
Comment (rdumusc) [inline rendering_resource_manager_service/config/models.py]: rm graceful_exit
exclusive = false by default if nb_cpus == 1 ?

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 2]
```diff
@@ -28,396 +28,378 @@
...
-import re
 import rendering_resource_manager_service.utils.custom_logging as log
 from rendering_resource_manager_service.session.models import \
     SESSION_STATUS_STARTING, SESSION_STATUS_RUNNING, \
     SESSION_STATUS_SCHEDULING, SESSION_STATUS_SCHEDULED
 import rendering_resource_manager_service.service.settings as global_settings
 
 
+SLURM_SSH_COMMAND = '/usr/bin/ssh -i ' + \
+                  global_settings.SLURM_KEY + ' ' + \
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py]: SLURM_KEY -> SLURM_SSH_KEY

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 3]
```diff
@@ -28,396 +28,378 @@
...
-        self._connected = False
         self._mutex = Lock()
 
-    def __connect(self):
+    def schedule(self, session, job_information):
         """
-        Utility method to connect to slurm queue, if not already done
+        Allocates a job and starts the rendering resource process. If successful, the session
+        job_id is populated and the session status is set to SESSION_STATUS_STARTING
+        :param session: The ID of the session
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py]: session: session for which the job should be scheduled

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 4]
```diff
@@ -28,396 +28,378 @@
...
+        return status
 
-    def schedule(self, session, job_information):
+    def allocate(self, session, job_information):
         """
-        Utility method to schedule an instance of the renderer on the cluster
+        Allocates a job according to rendering resource configuration. If the allocation is
+        successful, the session job_id is populated and the session status is set to
+        SESSION_STATUS_SCHEDULED
+        :param session: The ID of the session
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py]: idem

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 5]
```diff
@@ -28,396 +28,378 @@
...
-        return job.get_id()
-
-    def query(self, job_id):
+    def start(self, session, job_information):
         """
-        Verifies that a given job is up and running
-        :param job_id: The ID of the job
+        Start the rendering resource using the job allocated by the schedule method. If successful,
+        the session status is set to SESSION_STATUS_STARTING
+        :param session: The ID of the session
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py]: idem

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 6]
```diff
@@ -28,396 +28,378 @@
...
+            values += job_information.environment.split()
+            for variable in values:
+                full_command += variable + ' '
+
+            # Command lines parameters
+            rest_parameters = manager.RenderingResourceSettingsManager.format_rest_parameters(
+                str(rr_settings.scheduler_rest_parameters_format),
+                str(session.http_host),
+                str(session.http_port),
+                'rest' + str(rr_settings.id + session.id))
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py:197]: TODO remove later?

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 7]
```diff
@@ -28,396 +28,378 @@
...
-        response = json.dumps({'contents': msg})
-        return [400, response]
+                result = [400, response]
+        return result
 
-    @staticmethod
-    def job_information(session):
+    def hostname(self, job_id):
+        """
+        Retrieve the hostname for the batch host of the given job.
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py]: batch -> slurm

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 8]
```diff
@@ -28,396 +28,378 @@
...
+        :return: A string containing the status of the job
+        """
+        return self._query(job_id, 'JobStatus')
+
+    def rendering_resource_out_log(self, session):
+        """
+        Returns the contents of the rendering resource output file
         :param session: the session to be queried
-        :return: A string containing the information about the job
+        :return: A string containing the error log
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py]: error -> output

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 9]
```diff
@@ -28,396 +28,378 @@
...
+        return self._rendering_resource_log(session, settings.SLURM_ERR_FILE)
 
     @staticmethod
-    def rendering_resource_out_log(session):
+    def _query(job_id, attribute):
         """
-        Returns the contents of the rendering resource output file
+        Verifies that a given job is up and running
+        :param job_id: The ID of the job
+        :return: A Json response containing on ok status or a description of the error
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py]: typo on -> an

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 10]
```diff
@@ -28,396 +28,378 @@
...
+                log.error(str(e))
+        return value
+
+    @staticmethod
+    def _file_name(session, extension):
+        """
+        Returns the contents of the specified file
         :param session: the session to be queried
+        :param extension: file extension (typically err or out)
         :return: A string containing the error log
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py:374]: error log -> the filename

File: rendering_resource_manager_service/session/management/job_manager.py [hunk 11]
```diff
@@ -28,396 +28,378 @@
...
-        return globalJobManager.rendering_resource_log(session, settings.SLURM_OUT_FILE)
+        return settings.SLURM_OUTPUT_PREFIX + '_' + str(session.job_id) + \
+               '_' + session.renderer_id + '_' + extension
 
-    @staticmethod
-    def rendering_resource_err_log(session):
+    def _rendering_resource_log(self, session, extension):
         """
-        Returns the contents of the rendering resource error file
+        Returns the contents of the specified file
```
Comment (rdumusc) [inline rendering_resource_manager_service/session/management/job_manager.py:381]: file -> log extension

File: rendering_resource_manager_service/config/models.py (hunk 1 above)
Comment (favreau) [inline rendering_resource_manager_service/config/models.py]: Fixed

File: rendering_resource_manager_service/session/management/job_manager.py (hunk 2 above)
Comment (favreau) [inline rendering_resource_manager_service/session/management/job_manager.py]: Ok

File: rendering_resource_manager_service/session/management/job_manager.py (hunk 3 above)
Comment (favreau) [inline rendering_resource_manager_service/session/management/job_manager.py]: Ok

File: rendering_resource_manager_service/session/management/job_manager.py (hunk 4 above)
Comment (favreau) [inline rendering_resource_manager_service/session/management/job_manager.py]: Ok

File: rendering_resource_manager_service/session/management/job_manager.py (hunk 5 above)
Comment (favreau) [inline rendering_resource_manager_service/session/management/job_manager.py]: Ok

File: rendering_resource_manager_service/session/management/job_manager.py (hunk 7 above)
Comment (favreau) [inline rendering_resource_manager_service/session/management/job_manager.py]: Ok

File: rendering_resource_manager_service/session/management/job_manager.py (hunk 9 above)
Comment (favreau) [inline rendering_resource_manager_service/session/management/job_manager.py]: Ok

File: rendering_resource_manager_service/session/management/job_manager.py (hunk 11 above)
Comment (favreau) [inline rendering_resource_manager_service/session/management/job_manager.py:381]: Ok

**Comment by rdumusc** [comment]
+1

**Pull request merged**